src/libmongoc/tests/test-mongoc-topology.c

```diff
@@ -15,6 +16,123 @@
 #undef MONGOC_LOG_DOMAIN
 #define MONGOC_LOG_DOMAIN "topology-test"
 
+typedef struct {
+   int n_started;
+   int n_succeeded;
+   int n_failed;
+   int n_unknowns;
+   bson_mutex_t mutex;
```

Comment:
These counters are modified in SDAM callbacks and checked by the test. It wasn't thread-safe before and was showing up when running TSAN locally. Now, updating and reading them happens behind a mutex.